editor/editor_node.cpp

```diff
@@ -6103,11 +6103,8 @@ EditorNode::EditorNode() {
 	tabbar_container->add_child(scene_tabs);
 	distraction_free = memnew(Button);
 	distraction_free->set_flat(true);
-#ifdef OSX_ENABLED
-	distraction_free->set_shortcut(ED_SHORTCUT("editor/distraction_free_mode", TTR("Distraction Free Mode"), KEY_MASK_CMD | KEY_MASK_CTRL | KEY_D));
-#else
 	distraction_free->set_shortcut(ED_SHORTCUT("editor/distraction_free_mode", TTR("Distraction Free Mode"), KEY_MASK_CMD | KEY_MASK_SHIFT | KEY_F11));
-#endif
+	distraction_free->set_shortcut(ED_SHORTCUT("editor/distraction_free_mode.macOS", TTR("Distraction Free Mode"), KEY_MASK_CMD | KEY_MASK_CTRL | KEY_D));
```

Comment:
Ah damn, I just realised this won't work. The `distraction_free` button's shortcut will be overridden with the macOS option here, even if running on a different OS (e.g. Windows).

What would have to be done is that the ED_SHORTCUT be define elsewhere, even just before this statement, and then in the `set_shortcut` method, use `ED_GET_SHORTCUT`.

Reply:
Yeah defining it the line before and then using `GET` for the setter sounds good to me.